File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n<3)
          return 0;
          int count =0;
         vector<bool> sieve(n+1,true);
         for(int i =2;i<n;i++)
         {
            if(sieve[i])
            {
                count++;
                for(int j=2*i;j<=n;j+=i)
                {
                    sieve[j]= false;
                }
            }
         }
         return count;

    }
};

File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        if(n<2)
         return 0;
        boolean[] sieve = new boolean[n];
        Arrays.fill(sieve,true);
        sieve[0] = sieve[1] = false;
        for(int i =2;i*i<n;i++)
        {
            if(sieve[i])
            {
                for(int j=i*i;j<n;j+=i)
                    sieve[j]=false;
            }
        }
        int primeCount =0;
        for(boolean prime:sieve)
        {
            if(prime)
              primeCount++;
        }
        return primeCount;

    }
}